pkg/ctl/install/install.go

```diff
@@ -0,0 +1,448 @@
+		cmdutils.AddNameFlag(fs, rc.ClusterConfig.Metadata)
+		cmdutils.AddRegionFlag(fs, rc.ProviderConfig)
+		cmdutils.AddConfigFileFlag(fs, &rc.ClusterConfigFile)
+	})
+
+}
+
+type fluxInstaller struct {
+	opts          *installFluxOpts
+	resourceCmd   *cmdutils.ResourceCmd
+	k8sConfig     *clientcmdapi.Config
+	k8sRestConfig *rest.Config
+	k8sClientSet  *kubernetes.Clientset
+}
+
+func newFluxInstaller(rc *cmdutils.ResourceCmd, opts *installFluxOpts) (*fluxInstaller, error) {
+	if opts.templateParams.GitURL == "" {
+		return nil, fmt.Errorf("please supply a valid --git-url argument")
+	}
+	if opts.templateParams.GitEmail == "" {
+		return nil, fmt.Errorf("please supply a valid --git-email argument")
+	}
+
+	if err := cmdutils.NewMetadataLoader(rc).Load(); err != nil {
+		return nil, err
+	}
+	cfg := rc.ClusterConfig
+	ctl := eks.New(rc.ProviderConfig, cfg)
+	if !ctl.IsSupportedRegion() {
+		return nil, cmdutils.ErrUnsupportedRegion(rc.ProviderConfig)
+	}
+	if err := ctl.GetCredentials(cfg); err != nil {
+		return nil, err
+	}
+	kubernetesClientConfigs, err := ctl.NewClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+	k8sConfig := kubernetesClientConfigs.Config
+
+	k8sRestConfig, err := clientcmd.NewDefaultClientConfig(*k8sConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, errors.Errorf("cannot create Kubernetes client configuration: %s", err)
+	}
+	k8sClientSet, err := kubernetes.NewForConfig(k8sRestConfig)
+	if err != nil {
+		return nil, errors.Errorf("cannot create Kubernetes client set: %s", err)
+	}
+	fi := &fluxInstaller{
+		opts:          opts,
+		resourceCmd:   rc,
+		k8sConfig:     k8sConfig,
+		k8sRestConfig: k8sRestConfig,
+		k8sClientSet:  k8sClientSet,
+	}
+	return fi, nil
+}
+
+func (fi *fluxInstaller) run(ctx context.Context) error {
+	manifests, err := getFluxManifests(fi.opts.templateParams, fi.k8sClientSet)
+	if err != nil {
+		return fmt.Errorf("failed to create flux manifests: %s", err)
+	}
+
+	logger.Info("Cloning %s", fi.opts.templateParams.GitURL)
+	cloneDir, err := fi.cloneRepo(ctx)
+	if err != nil {
+		return fmt.Errorf("cannot clone repository %s: %s", fi.opts.templateParams.GitURL, err)
+	}
+	cleanCloneDir := false
+	defer func() {
+		if cleanCloneDir {
+			os.RemoveAll(cloneDir)
+		} else {
+			logger.Critical("You may find the local clone of %s used by eksctl at %s",
+				fi.opts.templateParams.GitURL,
+				cloneDir)
+		}
+	}()
+
+	logger.Info("Writing Flux manifests")
+	fluxManifestDir := filepath.Join(cloneDir, fi.opts.gitFluxPath)
+	if err := writeFluxManifests(fluxManifestDir, manifests); err != nil {
+		return err
+	}
+
+	if fi.opts.amend {
+		logger.Info("Stopping to amend the the Flux manifests, please exit the shell when done.")
+		if err := runShell(fluxManifestDir); err != nil {
+			return err
+		}
+	}
+
+	// If we need to create a Namespace, we do it synchronously before invoking kubectl apply.
```

Comment:
This shouldn't refer to kubectl anymore, since we are not using it

Reply:
Well spotted, I'll remove the reference to `kubectl` 👍 
About the race mentioned there, what would happen if we were to ensure that the namespace is first in the list of manifests we apply via `applyManifests`? Would we also experience such issues?

Reply:
> Well spotted, I'll remove the reference to `kubectl` 👍 

This part is done.
I'm still interested in more details about the race.

Reply:
FWIW, in https://github.com/weaveworks/eksctl/pull/1064/commits/1472ac818f459d3c26edd75fd4ea0bdfb7442b09 I've removed the bespoke logic we had under `createNamespaceSynchronously`. Now, the namespace is still created first, but using `CreateOrUpdate` like all other resources. This seems to work, but if you know it may not in some cases, please explain why (bonus points if links to relevant issues):
```console
$ EKSCTL_EXPERIMENTAL=true ./eksctl -f examples/01-simple-cluster.yaml install flux --git-url=git@github.com:marccarre/my-gitops-repo.git --git-email=carre.marc@gmail.com
[ℹ]  Cloning git@github.com:marccarre/my-gitops-repo.git
Cloning into '/var/folders/24/d3mml6bn20nftpt91cfldq1h0000gn/T/eksctl-install-flux-clone290012916'...
remote: Enumerating objects: 24, done.
remote: Counting objects: 100% (24/24), done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 24 (delta 2), reused 22 (delta 2), pack-reused 0
Receiving objects: 100% (24/24), 9.21 KiB | 4.60 MiB/s, done.
Resolving deltas: 100% (2/2), done.
[ℹ]  Writing Flux manifests
[ℹ]  Installing Flux into the cluster
[ℹ]  created "Namespace/flux"
[ℹ]  created "flux:ServiceAccount/flux"
[ℹ]  created "ClusterRole.rbac.authorization.k8s.io/flux"
[ℹ]  created "ClusterRoleBinding.rbac.authorization.k8s.io/flux"
[ℹ]  created "flux:Deployment.apps/flux"
[ℹ]  created "flux:Secret/flux-git-deploy"
[ℹ]  created "flux:Deployment.apps/memcached"
[ℹ]  created "flux:Service/memcached"
[ℹ]  Waiting for Flux to start
[ℹ]  Flux started successfully
[ℹ]  Committing and pushing Flux manifests to git@github.com:marccarre/my-gitops-repo.git
[master f9b49ac] Add Initial Flux configuration
 Author: Flux <carre.marc@gmail.com>
 6 files changed, 255 insertions(+)
 create mode 100644 flux/flux-account.yaml
 create mode 100644 flux/flux-deployment.yaml
 create mode 100644 flux/flux-secret.yaml
 create mode 100644 flux/memcache-dep.yaml
 create mode 100644 flux/memcache-svc.yaml
 create mode 100644 flux/namespace.yaml
Enumerating objects: 10, done.
Counting objects: 100% (10/10), done.
Delta compression using up to 8 threads
Compressing objects: 100% (9/9), done.
Writing objects: 100% (9/9), 3.89 KiB | 3.89 MiB/s, done.
Total 9 (delta 0), reused 8 (delta 0)
To github.com:marccarre/my-gitops-repo.git
   77abeff..f9b49ac  master -> master
[ℹ]  Flux will operate properly only once it has SSH access to: git@github.com:marccarre/my-gitops-repo.git
[ℹ]  please add the following Flux public SSH key to your repository:
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+Ej8uQcgl6soA4bTyVAktq/oOZo0QBpwBr0ZDk7Q6jY/SniJG4i4IHQHoJF914Lx6fPTtHVNy9brddcLupvsYih2GwqewD84RgwaaiGiDgf8Vey+RYaXj1zyRUTK4eFvTZ4KCgHTWzu/kDHraeq6jPrLfS0JkCbR/Ln3yLWQwqnPztUYoTGtPh14wb2biyN4QFqlBCHGUUS7YtT1MgCy8z8B6+3vnAL0h5U2a98SvHMXHkNvnvKetlCg619MFS2EfVrDoPSm650Ijmwnqp352vBdHOAGb5uq4PUnrXDzvAmfuadgSdpjiL1HHQJT+Uyi8zgunszwYS4wxVPXqJ3
```